packages/sync-service/lib/electric/connection/manager.ex

```diff
@@ -748,15 +767,19 @@ defmodule Electric.Connection.Manager do
         :replication_client_streamed_first_message,
         %State{current_phase: :running, current_step: :waiting_for_streaming_confirmation} = state
       ) do
-    # The call to `mark_connection_succeeded()` resets the backoff timer, so the next
-    # reconnection attempt will start from the minimum timeout and grow exponentially from
-    # there.
     # When the replication connection is stuck in a reconnection loop, we only mark it as
-    # having succeeded after receiving confirmation that streaming replication has started.
+    # having succeeded after receiving confirmation that streaming replication has started
+    # and waiting for some time to ensure no errors are sent over the stream due to a failure
+    # to start replication.
     # This is the only way to be sure because it can still fail after we issue the
     # start_streaming() call, so marking it as having succeeded earlier would result in a
     # reconnection loop with no exponential backoff.
-    state = mark_connection_succeeded(state)
+    Process.send_after(
+      self(),
+      {:replication_liveness_timeout, state.replication_client_pid},
+      @replication_liveness_confirmation_duration
+    )
```

Comment:
Just a nitpick: I would make this consistent with the other timers in this module and use the `:erlang.start_timer()` function. Its benefit is a uniform message format whereas with `send_after()` every message tends to be unique.

```suggestion
    :erlang.start_timer(
      @replication_liveness_confirmation_duration,
      self(),
      {:replication_liveness_check, state.replication_cliend_pid}
    )
```

and then in `handle_info` you'll need to match on

```
  def handle_info(
        {:timeout, _tref, {:replication_liveness_check, replication_client_pid}},
        %State{replication_client_pid: replication_client_pid} = state) do
...
```

On the technical side, there's barely any difference between the two, `Process.send_after()` also returns a timer ref.